aptos-move/vm-genesis/src/lib.rs

```diff
@@ -149,37 +153,39 @@ pub fn encode_aptos_mainnet_genesis_transaction(
     );
     initialize_features(
         &mut session,
+        &module_storage,
         genesis_config
             .initial_features_override
             .clone()
             .map(Features::into_flag_vec),
     );
-    initialize_aptos_coin(&mut session);
-    initialize_on_chain_governance(&mut session, genesis_config);
-    create_accounts(&mut session, accounts);
-    create_employee_validators(&mut session, employees, genesis_config);
-    create_and_initialize_validators_with_commission(&mut session, validators);
-    set_genesis_end(&mut session);
+    initialize_aptos_coin(&mut session, &module_storage);
+    initialize_on_chain_governance(&mut session, &module_storage, genesis_config);
+    create_accounts(&mut session, &module_storage, accounts);
+    create_employee_validators(&mut session, &module_storage, employees, genesis_config);
+    create_and_initialize_validators_with_commission(&mut session, &module_storage, validators);
+    set_genesis_end(&mut session, &module_storage);
 
     // Reconfiguration should happen after all on-chain invocations.
-    emit_new_block_and_epoch_event(&mut session);
+    emit_new_block_and_epoch_event(&mut session, &module_storage);
 
     let configs = vm.genesis_change_set_configs();
-    let (mut change_set, module_write_set) = session.finish(&configs).unwrap();
+    let (mut change_set, module_write_set) = session.finish(&configs, &module_storage).unwrap();
     assert_ok!(
         module_write_set.is_empty_or_invariant_violation(),
         "Modules cannot be published in this session"
     );
 
     // Publish the framework, using a different session id, in case both scripts create tables.
-    let state_view = GenesisStateView::new();
-    let resolver = state_view.as_move_resolver();
-
     let mut new_id = [0u8; 32];
     new_id[31] = 1;
     let mut session = vm.new_genesis_session(&resolver, HashValue::new(new_id));
-    publish_framework(&mut session, framework);
-    let (additional_change_set, module_write_set) = session.finish(&configs).unwrap();
+
+    let module_write_set = publish_framework(&mut session, &module_storage, framework);
+    let (additional_change_set, empty_module_write_set) =
+        session.finish(&configs, &module_storage).unwrap();
+    assert_ok!(empty_module_write_set.is_empty_or_invariant_violation());
```

Comment:
that's a bit convoluted way to assert (throwing away invariant violation) - we can just assert empty?
nit/personal preference also: i'd call it additional_module_write_set and assert it's empty, that way it describes what it is and assert describes the property that must hold.

Reply:
Addressed in #14402